src/cli/commands/install.js

```diff
@@ -69,6 +76,43 @@ type Flags = {
   tilde: boolean,
 };
 
+/**
+ * Try and detect the installation method for Yarn and provide a command to update it with.
+ */
+
+function getUpdateCommand(): ?string {
+  // Tarball install
+  if (fs2.existsSync(path.join(userHome, '.yarn'))) {
+    return 'yarn self-update';
+  }
+
+  // OSX
+  if (fs2.existsSync('/usr/local/Cellar')) {
+    return 'brew upgrade yarn';
+  }
+
+  // Debian
+  if (fs2.existsSync('/usr/share/lintian/overrides/yarn')) {
+    return 'sudo apt-get install yarn';
+  }
+
+  // npm
+  if (__dirname.indexOf('node_modules') >= 0) {
+    return 'npm upgrade --global yarn';
+  }
+
+  return null;
+}
+
+function getUpdateInstaller(): ?string {
+  // Windows
+  if (fs2.existsSync('C:/Program Files/Yarn') || fs2.existsSync('C:/Program Files (x86)/Yarn')) {
```

Comment:
This won't work as the user can select any directory during installation. Using the environment variable (`%ProgramFiles%\Yarn`) is better, but still not 100% accurate. Modifying `package.json` to have `"installationMethod": "msi"` as part of the build process would work best.